CONTRIBUTING.md

```diff
@@ -400,44 +406,72 @@ $ git commit -m "Back to -dev" phongo_version.h
 > "1.4.0beta1" and "beta" for the first beta release, this step would see them
 > ultimately changed to "1.4.0beta2-dev" and "devel".
 
-### Push commits and tags
-
-```
-$ git push
-$ git push --tags
-```
-
 ### Ensure Windows build artifacts exist
 
 Windows builds are tested by GitHub Actions. Each successful build for a pushed
 commit will produce a build artifact consisting of DLL and PDB files for that
 environment (e.g. 8.0-nts-x64). These build artifacts are later used to create
 release assets (see:
-[windows-release-build.yml])(.github/workflows/windows-release-build.yml).
+[windows-release-build.yml](.github/workflows/windows-release-build.yml)).
+
 Before publishing a release in GitHub, ensure that all Windows builds for the
 tag's commit have succeeded and that the necessary build artifacts have been
-created.
+created. This can be done by examining the build artifacts in the workflow run
+summary for the "Package X.Y.Z" commit (i.e. tag target).
 
-Note: the "published" event applies to both releases and pre-releases. See
-[Events that trigger workflows: release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)
-for more details.
+> **Note:** the "published" event applies to both releases and pre-releases. See
+> [Events that trigger workflows: release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)
+> for more details.
 
 ### Release PECL package
 
 The PECL package may be published via the
 [Release Upload](https://pecl.php.net/release-upload.php) form. You will have
 one chance to confirm the package information after uploading.
 
-### Merge the maintenance branch up to master
+### Branch management
+
+#### After releasing a new minor version
+
+After a new minor version is released (i.e. `master` was tagged), a maintenance
+branch should be created for future patch releases:
+
+```
+$ git checkout -b vX.Y
+$ git push mongodb vX.Y
+```
+
+Update `phongo_version.h` for the `master` branch:
+
+```diff
+-#define PHP_MONGODB_VERSION "1.15.1-dev"
++#define PHP_MONGODB_VERSION "1.16.0-dev"
+#define PHP_MONGODB_STABILITY "devel"
+-#define PHP_MONGODB_VERSION_DESC 1,15,1,0
++#define PHP_MONGODB_VERSION_DESC 1,16,0,0
+```
+
+Commit and push this change:
+
+```
+$ git commit -m "Master is now X.Y-dev" phongo_version.h
+$ git push mongodb
+```
+
+#### After releasing a patch version
+
+If this was a patch release, the maintenance branch must be merged up to master:
 
 ```
 $ git checkout master
+$ git pull mongodb master
 $ git merge vX.Y --strategy=ours
-$ git push
+$ git push mongodb
 ```
 
 The `--strategy=ours` option ensures that all changes from the merged commits
-will be ignored.
+will be ignored. This is OK because previously ensured that the `master` branch
+was up-to-date with all code changes in this maintenance branch before tagging.
```

Comment:
Thought it'd be helpful to remind readers why this is OK. Technically, we're just ignoring the `phongo_version.h` changes here.